File: sample-kotlin/src/main/java/com/king/mvvmframe/kotlin/bean/Result.java
package com.king.mvvmframe.kotlin.bean;

import com.google.gson.annotations.SerializedName;

/**
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 */
public class Result<T> {

    @SerializedName("error_code")
    private int code;

    private String message;

    @SerializedName("result")
    private T data;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public boolean isSuccess(){
        return  code == 0;
    }

    @Override
    public String toString() {
        return "Result{" +
                "code=" + code +
                ", message=" + message +
                ", data=" + data +
                '}';
    }
}

File: app/src/main/java/com/king/mvvmframe/data/model/Result.kt
package com.king.mvvmframe.data.model

import com.google.gson.annotations.SerializedName

/**
 * 结果
 *
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 * <p>
 * <a href="https://github.com/jenly1314">Follow me</a>
 */
data class Result<T>(
    @SerializedName("error_code")
    val code: Int,
    @SerializedName("reason")
    var message: String? = null,
    @SerializedName("result")
    var data: T? = null
) {
    /**
     * 是否成功
     */
    fun isSuccess() = code == 0
}
